src/Distributor.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.18;
+
+import {ECDSA} from "openzeppelin/utils/cryptography/ECDSA.sol";
+import {EIP712} from "openzeppelin/utils/cryptography/EIP712.sol";
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "openzeppelin/access/Ownable.sol";
+
+/// @title Elixir distributor
+/// @author The Elixir Team
+/// @custom:security-contact security@elixir.finance
+/// @notice Allows users to claim a token amount, approved by Elixir.
+contract Distributor is Ownable, EIP712 {
+    using SafeERC20 for IERC20;
+
+    /*//////////////////////////////////////////////////////////////
+                                VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Track claimed actions.
+    mapping(bytes32 => bool) public claimed;
```

Comment:
Is 32 bits going to be enough? That's 4 billion rewards, which sounds small. Any issues making this larger? We don't save anything by using a 32 over a 64 or even a 256, right?

Reply:
it's 256 bits, and is the digest hash, so the probability of generating the exact same hash is almost null

Reply:
I looked it up, here's the collision probability table for 32- and 64-bit numbers:
![image](https://github.com/ElixirProtocol/vertex-contracts/assets/3208589/8fb97aac-d1ce-4257-be5c-54e4df5a2777)

Reply:
yeah looks good, we are using 256 bits (32 bytes) -- so there are 2^256 possible hash values

Reply:
hash function with output size 256 bits, and we have p messages to hash, then probability of collision is about p^2/2^(n+1). For example, one billion messages (p=10^9) then the probability is about 4.3*10^-60.